xdem/volume.py

```diff
@@ -175,23 +233,45 @@ def calculate_hypsometry_area(ddem_bins: Union[pd.Series, pd.DataFrame], ref_dem
     return output
 
 
-def linear_interpolation(array: Union[np.ndarray, np.ma.masked_array]) -> np.ndarray:
+def linear_interpolation(array: Union[np.ndarray, np.ma.masked_array], max_search_distance: int = 10,
+                         extrapolate: bool = False, force_fill: bool = False) -> np.ndarray:
     """
-    Interpolate a 2D array using bilinear interpolation.
+    Interpolate a 2D array using rasterio's fillnodata.
```

Comment:
so is this `rasterio.fill.fillnodata` still a black box, or we now know a bit more what it does?!

Reply:
It's using [gdal_fillnodata](https://gdal.org/api/gdal_alg.html#_CPPv414GDALFillNodata15GDALRasterBandH15GDALRasterBandHdiiPPc16GDALProgressFuncPv). Reading the documentation, it actually looks to be an IDW approach. Or am I wrong??

Reply:
Yes, the documentation says inverse distance weighting, so that's what it should be.